lib/src/Uncertainty/Algorithm/OrthogonalBasis/OrthogonalUniVariatePolynomialFactory.cxx

```diff
@@ -55,7 +55,8 @@ OrthogonalUniVariatePolynomialFactory::OrthogonalUniVariatePolynomialFactory(con
   , recurrenceCoefficientsCache_(0)
   , polynomialsCache_(0)
 {
-  // Nothing to do. The derived class will have to call initializeCaches().
+  // The derived class will have to call initializeCaches().
+  if (!(measure.getDimension() == 1)) throw InvalidArgumentException(HERE) << "Error, expected a distribution of dimension 1, got dimension=" << measure.getDimension();
```

Comment:
You have changed your mind since you have proposed the **exact same** syntax in your PR related to NaNs in tests?

Reply:
If I did, it was a mistake on my part, but I do not see what you are referring to.

Reply:
I was thinking about your comment https://github.com/openturns/openturns/pull/1726#issuecomment-765364946

Reply:
Ah, ok. But in the case of `!=`, the "not" operator `!` is contained within `!=`, so I see no need to write `!(a==b)`. I think I used the `!=` syntax in my PR.

Reply:
Besides, in terms of warding NaNs off, `a!=b` achieves the same result as `!(a==b)`:

```
import numpy as np
a = np.nan
b = np.nan

# First syntax
if (!(a==b))
    raise ValueError("a must be equal to b")

# Second syntax
if (a!=b)
    raise ValueError("a must be equal to b")
```

Both syntaxes result in an error being raised.

Reply:
Once again I quote you:
```
The idea is to use the following structure for tests leading to an exception:
if (!good) throw Exception
instead of
if (bad) throw Exception
```
Here, good is for `measure.getDimension() == 1` and bad is for `measure.getDimension() != 1`.

So I propose that:
* you **clearly** define what is a **correct** way to write a test,
* you propose it as an addition to the coding rules https://openturns.github.io/openturns/latest/developer_guide/coding_rules.html
* Once it is accepted by the current active developers it becomes the rule to follow
In the mean time, if the code is correct it stays like it.

Reply:
OK, I will do that as part of my NaN-related PR. Clearly `!=` is a special case, because it cannot ever be a "good" condition.
If "a not equal to b" were the "good" condition, the check would need to be formulated as:
```
if (!( (a<b) || (b>a) )) throw Exception
```
That is because
```
if(!(a!=b)) throw Exception
```
is not NaN-robust. With, say, `a` being NaN, `a!=b` is True (regardless of `b`) and `!(a!=b)` is False, so the Exception is not thrown. That is why I think `!(a==b)` is not a good idea, although it *is* NaN-robust: it wrongly suggests that `!(a!=b)` is NaN-robust as well.